pydicer/convert/pet_to_nifti_convert_new.py

```diff
@@ -0,0 +1,409 @@
+        if (position_final_slice - axial_positions[-1]) / slice_spacing < 1.5:
+            # If only one slice is missing
+            diff = np.asarray([
+                not is_approx_equal(
+                    float(axial_positions[ind]) -
+                    float(axial_positions[ind - 1]) - slice_spacing, 0)
+                for ind in range(1, len(axial_positions))
+            ])
+            ind2interp = int(np.where(diff)[0])
+            new_slice = (np_image[:, :, ind2interp] +
+                         np_image[:, :, ind2interp + 1]) * 0.5
+            new_slice = new_slice[..., np.newaxis]
+            np_image = np.concatenate(
+                (np_image[..., :ind2interp], new_slice, np_image[...,
+                                                                 ind2interp:]),
+                axis=2)
+            warnings.warn(
+                "One slice is missing, we replaced it by linear interpolation")
+        else:
+            # if more than one slice are missing
+            raise RuntimeError('Multiple slices are missing')
+
+    image_position_patient = [float(k) for k in slices[0].ImagePositionPatient]
+    sitk_image = get_sitk_volume_from_np(np_image, pixel_spacing,
+                                         image_position_patient)
+
+    output_filepath = join(
+        output_folder,
+        correct_patient_name(str(slices[0].PatientName)) + '_' +
+        modality.lower() + extension)
+    sitk_writer.SetFileName(output_filepath)
+    sitk_writer.Execute(sitk_image)
+    if rtstruct_file is not None:
+        masks = get_masks(rtstruct_file,
+                          labels=labels_rtstruct,
+                          image_position_patient=image_position_patient,
+                          axial_positions=axial_positions,
+                          pixel_spacing=pixel_spacing,
+                          shape=np_image.shape,
+                          dtype=dtype_image)
+        for label, np_mask in masks:
+            output_filepath_mask = output_filepath.split(
+                '.')[0] + '_' + label.lower() + extension
+            sitk_mask = get_sitk_volume_from_np(np_mask, pixel_spacing,
+                                                image_position_patient)
+            sitk_writer.SetFileName(output_filepath_mask)
+            sitk_writer.Execute(sitk_mask)
+    return np.transpose(np_image,
+                        (1, 0, 2)), pixel_spacing, image_position_patient
+
+
+def get_sitk_volume_from_np(np_image, pixel_spacing, image_position_patient):
+    trans = (2, 0, 1)
+    sitk_image = sitk.GetImageFromArray(np.transpose(np_image, trans))
+    sitk_image.SetSpacing(pixel_spacing)
+    sitk_image.SetOrigin(image_position_patient)
+    return sitk_image
+
+
+def is_approx_equal(x, y, tolerance=0.05):
+    return abs(x - y) <= tolerance
+
+
+class MissingWeightException(RuntimeError):
+    pass
+
+
+def correct_patient_name(patient_name):

```

Comment:
This kind of thing is super specific to a particular dataset. We need to make sure the code in there in generalizable so that it can be used across different datasets.